src/widgets/modals/ByProgramModal.js

```diff
@@ -1,376 +1,248 @@
-/* eslint-disable react/destructuring-assignment */
 /* eslint-disable react/forbid-prop-types */
-/**
- * mSupply Mobile
- * Sustainable Solutions (NZ) Ltd. 2019
- */
-
-import React from 'react';
+import React, { useReducer, useEffect, useCallback, useMemo } from 'react';
```

Comment:
Nit nit: mixing default and deconstructed exports feels messy (just me?). Could do something like:

```
import React from 'react';
...
const { useReducer, ... } from React;
```